ls1intum/Artemis
# Improve exercise scores: performance, paging, searching, filtering

### Checklist

### Motivation and Context
Related to #999 

### Description
We use ngx-datatable to improve the performance when many scores are displayed. This enables pagination. All this is done on the client side.

What is missing:
- Pagination should be customizable, e.g. 50, 100, 200, 500, all
- Search functionality is needed so that users can find scores of specific students quickly

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate into a course
4. Click on Scores for a programming exercise (with many scores)
5. Check that the page is loaded quickly (less than 5s) and is rendered correctly. Sorting, filtering and searching should work

### Screenshots



![improved-datatable](https://user-images.githubusercontent.com/28230611/68041045-f7f57800-fccf-11e9-9f0d-cd684faae6f5.gif)
